vtkext/private/module/vtkF3DMetaImporter.cxx

```diff
@@ -0,0 +1,712 @@
+    if (localAnimationIndex < nAnim)
+    {
+      importer.first->DisableAnimation(localAnimationIndex);
+      return;
+    }
+    else
+    {
+      localAnimationIndex -= nAnim;
+    }
+  }
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DMetaImporter::IsAnimationEnabled(vtkIdType animationIndex)
+{
+  vtkIdType localAnimationIndex = animationIndex;
+  for (const auto& importer : this->Pimpl->Importers)
+  {
+    vtkIdType nAnim = importer.first->GetNumberOfAnimations();
+    if (localAnimationIndex < nAnim)
+    {
+      return importer.first->IsAnimationEnabled(localAnimationIndex);
+    }
+    else
+    {
+      localAnimationIndex -= nAnim;
+    }
+  }
+  return false;
+}
+
+//----------------------------------------------------------------------------
+vtkIdType vtkF3DMetaImporter::GetNumberOfCameras()
+{
+  return std::accumulate(this->Pimpl->Importers.begin(), this->Pimpl->Importers.end(), 0,
+    [](vtkIdType a, const auto& importerPair)
+    { return a + importerPair.first->GetNumberOfCameras(); });
+}
+
+//----------------------------------------------------------------------------
+std::string vtkF3DMetaImporter::GetCameraName(vtkIdType camIndex)
+{
+  vtkIdType localCameraIndex = camIndex;
+  for (const auto& importer : this->Pimpl->Importers)
+  {
+    vtkIdType nCam = importer.first->GetNumberOfCameras();
+    if (localCameraIndex < nCam)
+    {
+      std::string name = importer.first->GetCameraName(localCameraIndex);
+      if (name.empty())
+      {
+        name = "unnamed_" + std::to_string(camIndex);
+      }
+      return name;
+    }
+    else
+    {
+      localCameraIndex -= nCam;
+    }
+  }
+  return "";
+}
+
+//----------------------------------------------------------------------------
+void vtkF3DMetaImporter::SetCameraIndex(std::optional<vtkIdType> camIndex)
+{
+  this->Pimpl->CameraIndex = camIndex;
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DMetaImporter::GetTemporalInformation(vtkIdType animationIndex, double frameRate,
+  int& nbTimeSteps, double timeRange[2], vtkDoubleArray* timeSteps)
+{
+  vtkIdType localAnimationIndex = animationIndex;
+  for (const auto& importer : this->Pimpl->Importers)
+  {
+    vtkIdType nAnim = importer.first->GetNumberOfAnimations();
+    if (localAnimationIndex < nAnim)
+    {
+      return importer.first->GetTemporalInformation(
+        localAnimationIndex, frameRate, nbTimeSteps, timeRange, timeSteps);
+    }
+    else
+    {
+      localAnimationIndex -= nAnim;
+    }
+  }
+  return false;
+}
+
+//----------------------------------------------------------------------------
+bool vtkF3DMetaImporter::UpdateAtTimeValue(double timeValue)
+{
+  this->Pimpl->ColoringInfoUpdated = false;
+  bool ret = true;
+  for (const auto& importer : this->Pimpl->Importers)
+  {
+#if VTK_VERSION_NUMBER >= VTK_VERSION_CHECK(9, 3, 20240707)
+    ret &= importer.first->UpdateAtTimeValue(timeValue);
```

Comment:
It's not incorrect but I think bitwise operations on bools are usually just confusing.
```suggestion
    ret = ret && importer.first->UpdateAtTimeValue(timeValue);
```